packages/meteor/emitter-promise.js

```diff
@@ -0,0 +1,52 @@
+const { EventEmitter } = Npm.require('node:events');
+
+const DEFAULT_TIMEOUT =
+  Meteor.settings?.EMITTER_PROMISE_DEFAULT_TIMEOUT || 3000;
+
+/**
+ *
+ * @param emitter { EventEmitter } - EventEmitter used to trigger events emitter.emit("data", OBJ) or emitter.emit("error", ERROR_OBJ).
+ * @param timeout { Number } - Timout em ms to wait for events. Default 3000ms.
+ * @param onSuccess {function(data)} - Function called on succeceful receive data.
+ * @param onError {function(err)} - Function called when error or timeout occur.
+ * @returns {{promise: Promise<unknown>, emitter: EventEmitter}}
+ */
+const newPromiseResolver = ({
+  emitter = new EventEmitter(),
+  timeout = DEFAULT_TIMEOUT,
+  onSuccess,
+  onError,
+} = {}) => {
+  const promise = new Promise((resolve, reject) => {
+    const handler = setTimeout(() => {
+      emitter.emit(
+        'error',
+        new Meteor.Error(`EmitterPromise timeout: ${timeout}ms.`)
+      );
+    }, timeout);
+    emitter.once('data', (data) => {
+      clearTimeout(handler);
+      emitter.removeAllListeners();
+      resolve(data);
+      if (onSuccess) {
+        onSuccess(data);
+      }
+    });
+    emitter.once('error', (err) => {
+      clearTimeout(handler);
+      emitter.removeAllListeners();
+      reject(err);
+      if (onError) {
+        onError(err);
+      }
+    });
+  });
+  return {
+    emitter,
+    promise,
+  };
+};
+
+EmitterPromise = {
+  newPromiseResolver,
+};
```

Comment:
1.  I guess it was just the preference of who created it...
2. I'm not so adapted to TS. Are you willing to do it? Otherwise maybe @Grubba27 could take a look.

Reply:
2. It's not a must for now, and I'm definitely out of time for that 😐 But let's keep that in mind for the future - people from the community will probably pick it up.